ee/clickhouse/queries/funnels/funnel_correlation_persons.py

```diff
@@ -1,144 +1,189 @@
 
-        query, params = self.get_query()
-        results: List[Tuple[str]] = sync_execute(query, params)
 
-        return self._format_results(results)
+class _FunnelEventsCorrelationActors(ActorBaseQuery):
+    def __init__(self, filter: Filter, team: Team, base_uri: str = "/") -> None:
+        self._funnel_correlation = FunnelCorrelation(filter, team, base_uri=base_uri)
+        super().__init__(team, filter)
 
-    def get_query(self) -> Tuple[str, Dict[str, Any]]:
-        if self._filter.correlation_type == FunnelCorrelationType.PROPERTIES:
-            return self.get_properties_query()
+    @cached_property
+    def is_aggregating_by_groups(self) -> bool:
+        return self._filter.aggregation_group_type_index is not None
 
-        else:
-            return self.get_events_query()
+    def actor_query(self, extra_fields: Optional[List[str]] = None):
 
-    def get_properties_query(self) -> Tuple[str, Dict[str, Any]]:
+        if not self._filter.correlation_person_entity:
+            raise ValidationError("No entity for persons specified")
 
-        if not self._filter.correlation_property_values:
-            raise ValidationError("Property Correlation expects atleast one Property to get persons for")
+        assert isinstance(self._filter.correlation_person_entity, Entity)
 
         funnel_persons_query, funnel_persons_params = self._funnel_correlation.get_funnel_persons_cte()
 
+        prop_filters = self._filter.correlation_person_entity.properties
+        prop_query, prop_params = FunnelEventQuery(self._filter, self._team.pk)._get_props(prop_filters)
+
         conversion_filter = (
-            f'funnel_people.steps {"=" if self._filter.correlation_persons_converted else "<>"} target_step'
+            f'AND person.steps {"=" if self._filter.correlation_persons_converted else "<>"} target_step'
             if self._filter.correlation_persons_converted is not None
             else ""
         )
 
-        person_query, person_query_params = ClickhousePersonQuery(
-            self._filter,
-            self._team.pk,
-            entity=Entity({"id": "person", "type": "events", "properties": self._filter.correlation_property_values}),
-        ).get_query()
+        event_join_query = self._funnel_correlation._get_events_join_query()
 
         query = f"""
             WITH
                 funnel_people as ({funnel_persons_query}),
-                %(target_step)s AS target_step
+                toDateTime(%(date_to)s) AS date_to,
+                toDateTime(%(date_from)s) AS date_from,
+                %(target_step)s AS target_step,
+                %(funnel_step_names)s as funnel_step_names
             SELECT
-                DISTINCT funnel_people.person_id as person_id
-            FROM funnel_people
-            JOIN ({person_query}) person
-                ON person.id = funnel_people.person_id
-            WHERE {conversion_filter}
+                DISTINCT person.person_id as person_id
```

Comment:
While funnel_people needs to return data as person_id as a kludge we should solve, no reason we shouldn't `AS actor_id` here.